Add more direct connect sources

tomasdembelli opened the pull request:
Add following direct connect sources:
- Location
- Lag
- Customer Metadata
- LOA
- Hosted Connection
- Interconnect
- Router Configuration

Resolves #326

dylanratcliffe reviewed (changes requested):
Very nice. Couple of small suggestions.

One thing I just remembered, some of these APIs returns the tags directly while others require you to make another call. In this case the [DescribeTags](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/directconnect@v1.22.7#Client.DescribeTags) call. Can we please add that for the following resources that [support tags](https://docs.aws.amazon.com/directconnect/latest/UserGuide/using-tags.html), could you please add them:

* Connections
* Virtual interfaces
* Link aggregation groups (LAG)
* Interconnects
* Connections

tomasdembelli commented:
> Very nice. Couple of small suggestions.
> 
> One thing I just remembered, some of these APIs returns the tags directly while others require you to make another call. In this case the [DescribeTags](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/directconnect@v1.22.7#Client.DescribeTags) call. Can we please add that for the following resources that [support tags](https://docs.aws.amazon.com/directconnect/latest/UserGuide/using-tags.html), could you please add them:
> 
> * Connections
> * Virtual interfaces
> * Link aggregation groups (LAG)
> * Interconnects
> * Connections

@dylanratcliffe 

Thanks for this catch!

I've checked the links you've provided, and we are only missing tags for the direct connect gateway. For the other sources, the describe output object has the tags. We are populating them on to the sdp item.

For the direct connect gateway source, I've added a new function for getting the tags and updated the source accordingly.
See: https://github.com/overmindtech/aws-source/pull/383/commits/a67a07f8e3d25959d53d92fc418f5482e998d3ee
and: https://github.com/overmindtech/aws-source/pull/383/commits/5ca02b9b172722d06af7b7970786418db66947c6

Questions:
- I couldn't think of a way to test the new function unless we talk to the actual AWS endpoint. Any suggestions?
- Is the current implementation good? Is there any pattern already used in the other sources that I can inspire from?
- Do you want me to explicitly use this new call for the other sources that we get the tags from the describe output in their current implementation: connection, virtual interface, LAP, interconnect?

dylanratcliffe reviewed (approved):
Nah the solution you've come up with is a good one. If we are already getting the tags in the response no sense in making more requests than we need to. It's annoying how inconsistent this all is but not much to be done about it.

Regarding testing I was doing some playing around with [localstack](https://www.localstack.cloud/) last week which might be a good solution in that it would mock lots of the APIs that we need. But I haven't had a chance to look much further
